sklearn/linear_model/coordinate_descent.py

```diff
@@ -729,6 +737,9 @@ def fit(self, X, y, check_input=True):
         self.coef_, self.dual_gap_ = map(np.squeeze, [coef_, dual_gaps_])
         self._set_intercept(X_offset, y_offset, X_scale)
 
+        # workaround since _set_intercept will cast self.coef_ into float64
+        self.coef_ = np.asarray(self.coef_, dtype=X.dtype)
```

Comment:
But will this affect other models?

It would be safer to not touch `_preprocess_data` since it is called by lots of other models.
See [this comment](https://github.com/scikit-learn/scikit-learn/pull/6913#issuecomment-240020536).